### PR TITLE
#347: SelectionControlGroup: remove paddingY config

### LINKED ISSUE #347: SelectionControlGroup: remove paddingY config
This config was used to add a vertical padding before the first item, but we agreed it should always be half of the spacing between the options. So we can remove the config entry and just use `config.internalSpacing \ 2` as vertical padding.

We agreed to apply the same logic also to horizontal groups (the padding top will be half the horizontal spacing between options)

### PR DESCRIPTION
**Pull request by marcopiii**
Closes #347

### CODE REVIEW

File: packages/bento-design-system/src/CheckboxGroupField/createCheckboxGroupField.tsx [hunk 1]
```diff
@@ -74,8 +74,13 @@ export function createCheckboxGroupField(
             assistiveTextProps={descriptionProps}
             errorMessageProps={errorMessageProps}
           >
-            <Inset spaceY={config.group.paddingY}>
-              {(props.orientation || "vertical") === "vertical" ? (
+            <Inset
+              spaceY={
+                ((config.group.internalSpacing[props.orientation ?? "vertical"] as number) /
+                  2) as ComponentProps<typeof Inset>["spaceY"]
+              }
```
Comment (marcopiii) [inline packages/bento-design-system/src/CheckboxGroupField/createCheckboxGroupField.tsx:81]: @veej this fails at runtime if the `spacing/2` isn't a valid spacing, but allowing a custom spacing for `Inset` would make its interface less strict and I think it isn't something we want.

Comment (veej) [reply packages/bento-design-system/src/CheckboxGroupField/createCheckboxGroupField.tsx:81]: I think you can easily replace the `Inset` with a raw `Box` passing a `style` to bypass the design tokens

**Review by veej — APPROVED** [review]
Looks 👍

**Comment by veej** [comment]
😕 @marcopiii, looking at the diff highlighted by Chromatic, I think I make a mistake when I migrated the issue on GitHub. For horizontal groups, the vertical padding should still be `verticalInternalSpacing / 2`, instead of `horizontalInternalSpacing / 2`.
@Phyele could you confirm this is what we decided?

**Review by veej — APPROVED** [review]
Looks 👍

**Pull request merged**